[docs] Glossary additions and tweaks (#27162)
expo/expo: docs/pages/more/glossary-of-terms.mdx

A JavaScript function that is used to append [config mods](#config-mods) to the [app Config](#app-config) for use in [Prebuild](#prebuild). For more information, see [Config Plugins](/config-plugins/introduction/).

### Continuous Native Generation (CNG)

> An abstract concept that describes the process of generating ephemeral native projects from a set of inputs. The inputs are typically a combination of configuration and code, and the output is a native project that can be compiled into a native app. For more information, see [a CNG dedicated page](/workflow/continuous-native-generation/).

Reviewer: ```suggestion
An abstract concept that describes the process of generating ephemeral native projects from a set of inputs. The inputs are typically a combination of configuration and code, and the output is a native project that can be compiled into a native app. See [CNG](/workflow/continuous-native-generation/) guide for more information.
```

The process of generating the temporary native **android** and **ios** folders for a React Native project based on the [app config](#app-config). This process is performed by running the command `npx expo prebuild` from [Expo CLI](#expo-cli) in a project directory.

> See also: [Prebuild template](#prebuild-template) and [Autolinking](#autolinking).

Reviewer: ```suggestion
See [Prebuild template](#prebuild-template) and [Autolinking](#autolinking) for further information.
```

### React Native Web

> A high-performing abstraction on top of `react-dom` which enables core primitives from [React Native](#react-native) to run in the browser. React Native for web (also known as RNW) was developed at X (Twitter) and is currently used for their [main website](https://x.com). [Expo SDK](#expo-sdk) and [Expo CLI](#expo-cli) have first-class support for RNW.

Reviewer: ```suggestion
A high-performing abstraction on top of `react-dom` that enables core primitives from [React Native](#react-native) to run in the browser. React Native for web (RNW) was developed at X (Twitter) and is currently used for their [main website](https://x.com). [Expo SDK](#expo-sdk) and [Expo CLI](#expo-cli) have first-class support for RNW.
```
